test/user.test.js

```diff
@@ -1770,6 +1770,293 @@ describe('User', function() {
+          done();
+        });
+      };
+
+      it('keeps sessions AS IS if a new user is created using `create`', function(done) {
+        async.series([
+          function(next) {
+            User.create(
+              { 'email': 'newuser@example.com',
+                'password': 'newpass' },
+                function(err, newUserInstance) {
+                  if (err) return done(err);
+                  newUserCreated = newUserInstance;
+                  next();
+                });
+          },
+          function(next) {
+            User.login(
+              { 'email': 'newuser@example.com',
+                'password': 'newpass' },
+                function(err, newAccessToken) {
+                  if (err) return done(err);
+                  assert(newAccessToken.id);
+                  assertPreservedToken(next);
+                });
+          },
+        ], function(err) {
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if a new user is created using `updateOrCreate`', function(done) {
+        async.series([
+          function(next) {
+            User.create(
+              { 'email': 'newuser@example.com',
+                'password': 'newpass' },
+                function(err, newUserInstance2) {
+                  if (err) return done(err);
+                  newUserCreated = newUserInstance2;
+                  next();
+                });
+          },
+          function(next) {
+            User.login(
+              { 'email': 'newuser@example.com',
+                'password': 'newpass' },
+                function(err, newAccessToken2) {
+                  if (err) return done(err);
+                  assert(newAccessToken2.id);
+                  assertPreservedToken(next);
+                });
+          },
+        ], function(err) {
+          done();
+        });
+      });
+    });
+
+    describe('User not chaning email property', function() {
+      var user1, user2, user3;
+      it('preserves other users\' sessions if their email is  untouched', function(done) {
+        async.series([
+          function(next) {
+            User.create({ email: 'user1@example.com', password: 'u1pass' }, function(err, u1) {
+              if (err) return done(err);
+              User.create({ email: 'user2@example.com', password: 'u2pass' }, function(err, u2) {
+                if (err) return done(err);
+                User.create({ email: 'user3@example.com', password: 'u3pass' }, function(err, u3) {
+                  if (err) return done(err);
+                  user1 = u1;
+                  user2 = u2;
+                  user3 = u3;
+                  next();
+                });
+              });
+            });
+          },
+          function(next) {
+            User.login({ email: 'user1@example.com', password: 'u1pass' }, function(err, accessToken1) {
+              User.login({ email: 'user2@example.com', password: 'u2pass' }, function(err, accessToken2) {
```

Comment:
Unhandled `err` here and on the previous line too.

Why do we need so many users in this test? I mean why two users are not enough?